src/NRediSearch/SuggestionOptions.cs

```diff
@@ -0,0 +1,92 @@
+﻿// .NET port of https://github.com/RedisLabs/JRediSearch/
+
+namespace NRediSearch
+{
+    public class SuggestionOptions
+    {
+        private const string WITHPAYLOADS_FLAG = "WITHPAYLOADS";
+        private const string WITHSCORES_FLAG = "WITHSCORES";
+
+        public SuggestionOptions(SuggestionOptionsBuilder builder)
+        {
+            _with = builder._with;
+            Fuzzy = builder._fuzzy;
+            Max = builder._max;
+        }
+
+        public static SuggestionOptionsBuilder Builder => new SuggestionOptionsBuilder();
+
+        private With _with;
+        public With GetWith() => _with;
+
+        public bool Fuzzy { get; }
+
+        public int Max { get; } = 5;
+
+        public SuggestionOptionsBuilder ToBuilder() => new SuggestionOptionsBuilder(this);
+
+        public class With
```

Comment:
we know that java enums are always going to be slightly different to .NET; I wonder if this is just:

``` c#
[Flags]
public enum With {
    None = 0,
    Payloads = 1,
    Scores = 2,
    PayloadsAndScores = Payloads | Scores,
}
```

(and worry about the flags in a private method on `SuggestionOptions`)

Reply:
Fair enough.